### DISCUSSION TITLE
Toggle file_ignore_patterns

### DISCUSSION THREAD

**Discussion by rkperes** [ibhagwan/fzf-lua#1626] [General]
`file_ignore_patterns` is great, yet sometimes I'd like to easily disable it, like we already have the default configuration for .gitignore in `files`.

How could one achieve so? I tried a custom action using `actions.toggle_action`, but I understood that's not targeting the configuration object.

Is there a way to achieve that? 
I also understand there's the additional layering complication, where `file_ignore_patterns` is a result of merging tables from different levels of the configuration, and delivering a default option for that might not be straight-forward.
I'd be happy with just a workaround for my setup, where I set `file_ignore_patterns` only at the inner level of the config, and I'm abble to toggle (enable/disable) the option via some keymap.

**Comment by ibhagwan** [comment]


**Reply by ibhagwan** [reply]
How about this?
```lua
local _ignore = false
_G.my_files = function()
  _ignore = not _ignore
  local opts = {
    file_ignore_patterns = _ignore == false and false or nil,
    actions = {
      ['ctrl-g'] = {
        fn = function() _G.my_files() end,
        desc = 'foo',
        header = function() return _ignore == false and 'enable' or 'disable' end,
      }
    }
  }
  require('fzf-lua').files(opts)
end
```

**Reply by nyngwang** [reply]
@ibhagwan
In #1870 I'm trying to create a toggle action of a similar form here, but instead, I want to toggle those "entries" to be filtered between _a set_(e.g. `{1,2,3,...,100}`) and its _subset_(e.g. `{1,2,3,...,50}`). Is this possible? I need some guidance. My apologies if my direct tagging is annoying.

**Reply by ibhagwan** [reply]
> @ibhagwan In #1870 I'm trying to create a toggle action of a similar form here, but instead, I want to toggle those "entries" to be filtered between _a set_(e.g. `{1,2,3,...,100}`) and its _subset_(e.g. `{1,2,3,...,50}`). Is this possible? I need some guidance. My apologies if my direct tagging is annoying.

Use similar code to what I posted above your comment, instead of toggling and toggle between two different file_ignore_ patterns with  lua patterns matching your set/subset.

**Reply by rkperes** [reply]
I never came back to thank you, but thanks a lot. Here's what I'm using now with a few tweaks:
* I had to change it to use the `file_ignore_patterns` to set the proper config when enabled. This was a required fix, AFAIU.
* Refactor so I could wrap different fzf functions that support the config.
* I changed it to use local variables only. That was a matter of preference, not really required.

```lua
local file_ignore_patterns = {
  -- set the patterns to ignore here, as you'd use the in the standard configuration
  '/test[^/]*/',
  '/[^/]*test/',
  '_test.go$',
  '_templ.go$',
  'BUILD.bazel',
  '^\\..*',
}

local fzf_custom_ignore_func
fzf_custom_ignore_func = function(fzf_func, ignore, flip_ignore)
  local fip = ignore == true and file_ignore_patterns or nil
  local opts = {
    file_ignore_patterns = fip,
    actions = {
      ['ctrl-g'] = {
        fn = function()
          ignore = flip_ignore()
          fzf_custom_ignore_func(fzf_func, ignore, flip_ignore)
        end,
        desc = 'Ignore custom file set',
        header = function()
          return ignore == false and 'hide ignored files' or 'show ignored files'
        end,
      },
    },
  }
  fzf_func(opts)
  return ignore
end

local fzf_func_wrap_with_ignore = function(fzf_func, initial)
  local ignore = initial
  local flip_ignore = function()
    ignore = not ignore
    return ignore
  end

  return function()
    fzf_custom_ignore_func(fzf_func, ignore, flip_ignore)
  end
end
```

This allows me to reuse it in several functions, each with its separate config state and default:
```lua
vim.keymap.set('n', '<leader>ff', fzf_func_wrap_with_ignore(require('fzf-lua').files, true), { desc = 'Files' })
vim.keymap.set('n', '<leader>fg', fzf_func_wrap_with_ignore(require('fzf-lua').grep_project, true), { desc = 'Grep' })
```

**Comment by dennypenta** [comment]
file_ignore_patterns = false helped me to configure live grep in ~/.config, but I couldn't find it documented.
would it be useful to cover it in fzf-lua.txt?

**Reply by ibhagwan** [reply]
It's automatically generated from the README, it's somewhat specific, maybe better to add it to OPTIONS.md? or the Wiki (which is open edit if you wish to contribute)?
